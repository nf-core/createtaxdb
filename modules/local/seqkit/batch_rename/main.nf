process SEQKIT_BATCH_RENAME {
    tag "${meta.id}"
    label 'process_low'

    conda "${moduleDir}/environment.yml"
    container "${workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container
        ? 'https://depot.galaxyproject.org/singularity/seqkit:2.9.0--h9ee0642_0'
        : 'biocontainers/seqkit:2.9.0--h9ee0642_0'}"

    input:
    tuple val(meta), path(fastx), path(replace_tsv)

    output:
    tuple val(meta), path("renamed_files/*"), emit: fastx
    path "versions.yml", emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''

    """
    while IFS=\$'\\t' read -r fasta_name replace_str out_fname ; do
        seqkit \\
            replace \\
            ${args} \\
            --threads ${task.cpus} \\
            -i \$fasta_name \\
            -r \$replace_str \\
            -o renamed_files/\$out_fname \\
    done < ${replace_tsv}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        seqkit: \$( seqkit version | sed 's/seqkit v//' )
    END_VERSIONS
    """

    stub:
    def prefix = task.ext.prefix ?: "${meta.id}"
    def extension = "fastq"
    if ("${fastx}" ==~ /.+\.fasta|.+\.fasta.gz|.+\.fa|.+\.fa.gz|.+\.fas|.+\.fas.gz|.+\.fna|.+\.fna.gz/) {
        extension = "fasta"
    }
    def endswith = task.ext.suffix ?: "${extension}.gz"

    """
    echo "" | gzip > ${prefix}.${endswith}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        seqkit: \$( seqkit version | sed 's/seqkit v//' )
    END_VERSIONS
    """
}
