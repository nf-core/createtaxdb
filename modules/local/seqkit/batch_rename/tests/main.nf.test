nextflow_process {

    name "Test Process SEQKIT_BATCH_RENAME"
    script "../main.nf"
    process "SEQKIT_BATCH_RENAME"

    tag "modules"
    tag "modules_nfcore"
    tag "seqkit"
    tag "seqkit/batch_rename"

    test("sarscov2 - fasta - replace") {

        when {
            process {
                """
                    input[0] = [
                        [ id:'test' ],   // meta map
                        [ file('https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/genomics/sarscov2/genome/proteome.fasta.gz'), file('https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/genomics/prokaryotes/bacteroides_fragilis/genome/genome.fna.gz') ],
                    ]
                    input[1] = Channel.of('proteome.fasta.gz\t^.*\$\t1234\tproteome.fasta', 'genome.fna.gz\t^.*\$\t1234\tgenome.fna').toList()

                """
            }
        }
//

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("sarscov2 - fasta - stub") {

        options "-stub"

        when {
            process {
                """
                    input[0] = [
                        [ id:'test' ],   // meta map
                        [ file('https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/genomics/sarscov2/genome/proteome.fasta.gz'), file('https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/genomics/prokaryotes/bacteroides_fragilis/genome/genome.fna.gz') ],
                    ]
                    input[1] = Channel.of('proteome.fasta.gz\t^.*\$\t1234\tproteome.fasta', 'genome.fna.gz\t^.*\$\t1234\tgenome.fna').toList()

                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }
}
