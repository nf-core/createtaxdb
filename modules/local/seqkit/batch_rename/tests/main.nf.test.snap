{
    "sarscov2 - fasta - replace": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            "genome.fna:md5,1a3cad6103ec69ea52bca9d46b6aa478",
                            "proteome.fasta:md5,db0ecd5dbce6bf9730685b94ec87854d"
                        ]
                    ]
                ],
                "1": [
                    "versions.yml:md5,8d27e93173ef8d4941faef6af25ba3d7"
                ],
                "fastx": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            "genome.fna:md5,1a3cad6103ec69ea52bca9d46b6aa478",
                            "proteome.fasta:md5,db0ecd5dbce6bf9730685b94ec87854d"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,8d27e93173ef8d4941faef6af25ba3d7"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-05-23T15:41:34.273717"
    }
}